### PR TITLE
Add CentOS 8 support and remove CentOS 6

### PR DESCRIPTION
**Pull request by enrico-usai**
This patch is a rebase and cleanup of the following patches:

* Add CentOS 8 support #2084
* Update templates for Centos 8 #2109
* Centos8: Fix yum proxy setting #2128
* Deprecate Centos6 #2182
* Extend RAID test to work with different mdadm versions #2184

### CODE REVIEW

**Review by rexcsn — APPROVED** [review]
LGTM!🚀
But looks like code check is failing

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ CHANGELOG
 
 **ENHANCEMENTS**
 
+- Add support for CentOS 8
```
Comment (demartinofra) [inline CHANGELOG.md:9]: shall we add: 'in all Commercial and GovCloud regions'?

Comment (lukeseawalker) [reply CHANGELOG.md]: probably just in global regions (commercial)

**Review by rexcsn — DISMISSED** [review]
LGTM!

File: CHANGELOG.md (hunk 1 above)
Comment (enrico-usai) [reply CHANGELOG.md]: Fixed

**Review by demartinofra — APPROVED** [review]

**Pull request merged**